File: src/handlers/smsHandler.js
/**
 * SMS Handler
 * 處理 type: sms 的請求
 * 
 * Payload 結構:
 * {
 *   phone: string,   // 收件者電話號碼
 *   message: string  // 簡訊內容
 * }
 */

/**
 * 驗證 SMS payload
 * @param {Object} payload
 * @returns {Object} { valid, error? }
 */
const validatePayload = (payload) => {
  const { phone, message } = payload;

  if (!phone) {
    return { valid: false, error: '缺少 phone 欄位' };
  }

  if (!message) {
    return { valid: false, error: '缺少 message 欄位' };
  }

  return { valid: true };
};

/**
 * 發送 SMS
 * @param {Object} payload - SMS 內容
 */
const send = (payload) => {
  const { phone, message } = payload;

  // TODO: 實作 sms 發送邏輯
  console.log('Sending SMS to:', phone);

  // 產生唯一 ID
  const id = `sms_${Date.now()}`;

  return {
    success: true,
    message: 'SMS 已排入發送佇列',
    data: {
      id,
      status: 200
    }
  };
};

module.exports = {
  validatePayload,
  send
};

File: internal/handlers/sms_handler.go
package handlers

import (
	"errors"
	"fmt"
	"time"

	"mps/internal/processor"
)

// SMSHandler handles sms-type messages.
type SMSHandler struct{}

// NewSMSHandler creates a new SMSHandler instance.
func NewSMSHandler() *SMSHandler {
	return &SMSHandler{}
}

// Type returns the message type key for SMSHandler.
func (h *SMSHandler) Type() string {
	return "sms"
}

// Handle validates required sms payload fields and returns a mocked send result.
// Required fields are to and message.
func (h *SMSHandler) Handle(payload map[string]any) (*processor.HandlerResult, error) {
	to, ok := payload["to"].(string)
	if !ok || to == "" {
		return nil, errors.New("invalid sms payload: to is required")
	}

	message, ok := payload["message"].(string)
	if !ok || message == "" {
		return nil, errors.New("invalid sms payload: message is required")
	}

	_ = to
	_ = message

	return &processor.HandlerResult{
		Message: "SMS sent successfully",
		Data: map[string]any{
			"messageId": fmt.Sprintf("msg-%d", time.Now().UnixNano()),
			"status":    "sent",
		},
	}, nil
}
